### DISCUSSION TITLE
nu lsp not working in nixos

### DISCUSSION THREAD

**Discussion by kyp0717** [nushell/nushell#14414] [General]
Hello.  I can't seems to start up the nu lsp using the command `nu --lsp` .
How to check if it is running.
I tried this command `ps | where name =~ "nu"` and these are the processes that are found ..
![image](https://github.com/user-attachments/assets/84823f73-2a31-4690-8d14-7b6e41b971b0)

What am I doing wrong?

**Comment by fdncred** [comment]
I can tell it's working by the functionality. e.g. completions in nvim.
![image](https://github.com/user-attachments/assets/96cca688-167c-4da1-8800-f12fbf86b0a0)

**Comment by ghost** [comment]
Im not sure what you are trying to do, but i think you might have misunderstoop what a LSP is.  
LSP is a confusing name due to the `server` in its name. In 99% of cases you do not have to start a LSP-server. Instead your editor will launch a nu instance on its own and use that as "server".

[explanation of what a LSP is](https://en.wikipedia.org/wiki/Language_Server_Protocol)

If you just run `nu --lsp` in your terminal it should just do nothing (and wait for stdin input and then crash if you type in something since you don't know the weird format).

as to how to use it: choose a editor with LSP support (or a plugin for it) and look at its documentation. Examples of compatible editors: (neo-)vim, and helix

**Reply by kyp0717** [reply]
Thank you for the clarification.  This explanation helps me understand LSP better.  The reason I raised this issue was that when I integrate nu lsp with neovim as a plugin to detect nu filetype, it was not successful.  I can tell it did not work because `LspInfo` indicated that an LSP client was not started for the nu filetype.  I follow the guideline provided in nvim-lspconfig for nu.

But thank you for the responds thus far.  I will keep trying and get back to this based on the responds provided thus far.

**Reply by ghost** [reply]
i haven't used neovim for a while, but i think you had to manually specify the path the `nu` binary since for it to work

EDIT: i checked my config: i used <https://github.com/zioroboco/nu-ls.nvim> since the `lsp-config` one didn't work for me at the time. i used <https://github.com/LhKipp/nvim-nu> for syntax-highlighting, but AFAIK it now is a good and active all-in-one nushell config (including LSP).

**Reply by fdncred** [reply]
I followed this as a guide https://github.com/nushell/integrations/blob/main/nvim/init.lua
